cjdns-core/src/announcement/mod.rs

```diff
@@ -0,0 +1,16 @@
+//! Announcement message
+//! `Ann` or `ann` are shorthands for `Announcement` and `announcement`.
+
+pub use models::*;
+pub use serialized_ann::serialized_data::*;
+
+mod serialized_ann;
+mod models;
+mod errors;
+
+
+// todo
+// 1. Reformat Errors (because we have mix of inner announcement errors and errors from external crates) in serialized_ann module in the next way:
+//   - all the functions except for pub should return Box<dyn std::error::Error> | Result<U, &'static str>
+//   - but main pub functions (like `parser::parse`, `serialized_data::AnnouncementPacket::{parse, check}` should return original errors with .or() | map_err
```

Comment:
In the end better to use some error handling library in the end (like Thiserror or Anyhow).
For now can be kept as is.